File: Introibo/Data/Course.swift
import Foundation

// Matches Introibo/Resources/courses.json.
// Ported from prototype/learn.html's COURSES object.

struct Course: Identifiable, Decodable, Hashable {
    let slug: String
    let num: Int
    var title: String
    let latin: String
    var intro: String
    var sections: [Section]

    var id: String { slug }

    struct Section: Decodable, Hashable {
        let type: String         // lesson | tip | cards | summary | phrase | table
        var label: String?
        var html: String?        // present for lesson/tip/summary/phrase/table
        var note: String?        // present for cards
        var items: [Card]?       // present for cards

        struct Card: Decodable, Hashable {
            let lat: String?
            var phon: String?
            var eng: String?
        }
    }
}

File: android/app/src/main/java/com/lampstandhq/introibo/data/model/Course.kt
package com.lampstandhq.introibo.data.model

import kotlinx.serialization.Serializable

/**
 * Matches courses.json.
 */
@Serializable
data class Course(
    val slug: String,
    val num: Int,
    val title: String,
    val latin: String,
    val intro: String,
    val sections: List<Section>,
) {
    @Serializable
    data class Section(
        val type: String,            // lesson | tip | cards | summary | phrase | table
        val label: String? = null,
        val html: String? = null,    // present for lesson/tip/summary/phrase/table
        val note: String? = null,    // present for cards
        val items: List<Card>? = null, // present for cards
    ) {
        @Serializable
        data class Card(
            val lat: String? = null,
            val phon: String? = null,
            val eng: String? = null,
        )
    }
}
